### ISSUE TITLE
Builds fail due to linting error

### ISSUE THREAD

**Issue by berryp** [zoni/obsidian-export#149]
Example from: https://github.com/zoni/obsidian-export/actions/runs/3933231631/jobs/6726713876

```
Run actions-rs/cargo@v1
/home/runner/.cargo/bin/cargo clippy -- -D warnings
    Checking obsidian-export v22.11.0 (/home/runner/work/obsidian-export/obsidian-export)
error: the borrowed expression implements the required traits
Error:    --> src/lib.rs:384:58
    |
384 |                 let destination = &self.destination.join(&relative_path);
    |                                                          ^^^^^^^^^^^^^^ help: change this to: `relative_path`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
    = note: `-D clippy::needless-borrow` implied by `-D warnings`

error: the borrowed expression implements the required traits
Error:    --> src/lib.rs:[7](https://github.com/zoni/obsidian-export/actions/runs/3933231631/jobs/6726713876#step:8:8)35:50
    |
735 |             || path_normalized_lowered.ends_with(&filename_normalized.to_lowercase())
    |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: change this to: `filename_normalized.to_lowercase()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

error: could not compile `obsidian-export` due to 2 previous errors
Error: The process '/home/runner/.cargo/bin/cargo' failed with exit code [10](https://github.com/zoni/obsidian-export/actions/runs/3933231631/jobs/6726713876#step:8:11)1
```

**Comment by berryp** [comment]
Moved the PR to draft for now as it looks like a rabbit hole (more lint failures). I'll come back to this later.

**Comment by epsilonhalbe** [comment]
should be fixed in PR #163

**Comment by zoni** [comment]
`main` is passing again :smile:
